SU2_CFD/src/fluid/CSU2TCLib.cpp

```diff
@@ -1829,227 +1946,134 @@ void CSU2TCLib::DiffusionCoeffGY(){
-  // }
 }
 
 void CSU2TCLib::ThermalConductivitiesGY(){
 
-  su2double Mi, Mj, mi, mj, gam_i, gam_j, denom_t, denom_r, d1_ij, d2_ij, a_ij, Omega_ij;
-
-  su2double pi   = PI_NUMBER;
-  su2double Na   = AVOGAD_CONSTANT;
-  su2double kb   = BOLTZMANN_CONSTANT;
-
-  if (ionization) {
-    SU2_MPI::Error("NEEDS REVISION w/ IONIZATION",CURRENT_FUNCTION);
-  }
+  const su2double Na   = AVOGAD_CONSTANT;
+  const su2double kb   = BOLTZMANN_CONSTANT;
 
   /*--- Mixture vibrational-electronic specific heat ---*/
-  Cvves = ComputeSpeciesCvVibEle(Tve);
+  const auto Cvves = ComputeSpeciesCvVibEle(Tve);
   su2double rhoCvve = 0.0;
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
     rhoCvve += rhos[iSpecies]*Cvves[iSpecies];
-  su2double Cvve = rhoCvve/Density;
+  const su2double Cvve = rhoCvve/Density;
 
   /*--- Calculate mixture gas constant ---*/
   su2double R = 0.0;
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
-    R += Ru * rhos[iSpecies]/Density;
+    R += Ru / MolarMass[iSpecies] * rhos[iSpecies]/Density;
   }
 
   /*--- Mixture thermal conductivity via Gupta-Yos approximation ---*/
-  ThermalCond_tr = 0.0;
-  ThermalCond_ve = 0.0;
+  su2double ThermalCond_tr = 0.0;
+  su2double ThermalCond_ve = 0.0;
   for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
 
     /*--- Calculate molar concentration ---*/
-    Mi      = MolarMass[iSpecies];
-    mi      = Mi/Na;
-    gam_i   = rhos[iSpecies] / (Density*Mi);
-    denom_t = 0.0;
-    denom_r = 0.0;
+    const su2double Mi    = (MolarMass[iSpecies] + EPS);
+    const su2double mi    = Mi/Na;
+    const su2double gam_i = rhos[iSpecies] / (Density*Mi);
+    su2double denom_t = 0.0;
+    su2double denom_r = 0.0;
+    su2double denom_re = 0.0;
+
     for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
-      Mj    = MolarMass[jSpecies];
-      mj    = Mj/Na;
-      gam_j = rhos[iSpecies] / (Density*Mj);
-      a_ij = 1.0 + (1.0 - mi/mj)*(0.45 - 2.54*mi/mj) / ((1.0 + mi/mj)*(1.0 + mi/mj));
+      const su2double Mj    = (MolarMass[jSpecies] + EPS);
+      const su2double mj    = Mj/Na;
+      const su2double gam_j = rhos[iSpecies] / (Density*Mj);
+      const su2double a_ij = 1.0 + (1.0 - mi/mj)*(0.45 - 2.54*mi/mj) / ((1.0 + mi/mj)*(1.0 + mi/mj));
 
-      /*--- Calculate the Omega^(0,0)_ij collision cross section ---*/
-      Omega_ij = 1E-20 * Omega00(iSpecies,jSpecies,3)
-          * pow(T, Omega00(iSpecies,jSpecies,0)*log(T)*log(T)
-          + Omega00(iSpecies,jSpecies,1)*log(T)
-          + Omega00(iSpecies,jSpecies,2));
+      const su2double T_col = ((iSpecies == 0 && ionization) || (jSpecies == 0 && ionization)) ? Tve : T; 
 
-      /*--- Calculate "delta1_ij" ---*/
-      d1_ij = 8.0/3.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*T*(Mi+Mj))) * Omega_ij;
+      su2double d1_ij = ComputeCollisionDelta(iSpecies, jSpecies, Mi, Mj, T_col, true);
+      su2double d2_ij = ComputeCollisionDelta(iSpecies, jSpecies, Mi, Mj, T_col, false);
+      if (d1_ij >1E16) d1_ij = 1E16;
```

Comment:
Is there a better way to check for Inf?   In some cases d1 or d2 can be infinity but this causes NaNs.